### PR TITLE
Replace all artisanal threading code with a Celluloid based implementation.

### PR DESCRIPTION
**Pull request by alloy**
As requested in #2 and to ensure we can safely run long-running connections.

This also adds support for connection pools, is self-healing, and has been thoroughly tested for thread safety and connection issues.

/cc @dblock @joeyAghion

### CODE REVIEW

File: examples/long-running.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                                    keep_alive: true) # This option is the key to long running connections
+client.connect
+
+loop do
+  begin
+    logger.info "Perform burst"
+    # Perform a burst of notifications from multiple concurrent threads to demonstrate thread safety.
+    #
+    Array.new(3) do
+      Thread.new do
```
Comment (dblock) [inline examples/long-running.rb]: If you're going to use Celluloid, then use its support for concurrent tasks too, not `Thread`.

File: examples/long-running.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        client.group do |group|
+          10.times do
+            notification = Lowdown::Notification.new(:token => device_token)
```
Comment (dblock) [inline examples/long-running.rb]: Rubocop this! `token: device_token` ;)

**Comment by dblock** [comment]
:+1:

File: examples/long-running.rb (hunk 1 above)
Comment (alloy) [inline examples/long-running.rb]: This is an example file that shows how people using the library might typically use it, this using plain Thread should better illustrate that imo.

Comment (alloy) [inline examples/long-running.rb]: To be clear, the actual connection objects do use Celluloid's facilities for concurrency.

File: examples/long-running.rb (hunk 2 above)
Comment (alloy) [inline examples/long-running.rb]: Good point, added rubocop. However… https://github.com/alloy/lowdown/blob/6477b1338e0246adff725a42f094da612f26ec3a/.rubocop.yml#L96-L99 ;)

**Pull request merged**